sgkit/tests/test_distance.py

```diff
@@ -0,0 +1,45 @@
+import dask.array as da
+import numpy as np
+import pytest
+from scipy.spatial.distance import pdist as scipy_pdist  # type: ignore
+from scipy.spatial.distance import squareform
+
+from sgkit.distance.api import pdist
+from sgkit.typing import ArrayLike
+
+
+def get_vectors(array_type: str = "da") -> ArrayLike:
+    if array_type == "da":
+        rs = da.random.RandomState(0)
+        x = rs.randint(0, 3, size=(18, 40), dtype="i1").rechunk((6, 10))
+    else:
+        x = np.random.rand(18, 40)
+    return x
+
+
+def test_distance_correlation() -> None:
+    x = get_vectors()
+    distance_matrix = pdist(x, metric="correlation")
+    np.testing.assert_almost_equal(distance_matrix, np.corrcoef(x).compute())
+
+
+def test_distance_euclidean() -> None:
```

Comment:
I think missing values and chunking are two things that are worth adding to these tests (probably as parameters).

Reply:
Oh also that float32 inputs result in float32 output arrays.

Reply:
I think its probably worth having separate tests for each instead of parameters, so that when it fails we know, what exactly has failed?

Reply:
Pytest makes it obvious which parameter a failure is associated with so I wouldn't worry about it much.  All of the other "dtype preservation" tests use parameters as well so my weak preference would be for parameters.

Reply:
Never mind, I have already [parameterised most of the them](https://github.com/pystatgen/sgkit/pull/306/files#diff-bfebf015a7d0a911d66743edcf9806a15b4661814667c0683fef0af8a6222b35R111), let me know if that doesn't makes sense.